File: Boletines/Boletin2/Ejercicio 4/nodoB/src/nodo_B.py
#!/usr/bin/env python3

from tkinter import N
import rospy
from std_msgs.msg import Int32

i = 0
suma = 0
numero = 0  

def callback(mensaje):
    """
    Funcion para obetener el valor del mensaje
    """
    global i, suma , respuesta
    i += 1
    numero = mensaje.data
    
    suma += numero
    print(numero ,"La suma es: ", suma, "I", i)
    
    if i == 10:
        print("condicion cumplida")
        respuesta.publish(suma)
        i = 0
        suma = 0


rospy.init_node('nodo_B')
rospy.loginfo('Nodo B iniciado')


#Nos subcribimos al nodo 
rospy.Subscriber('nodo_A', Int32, callback)

print("La suma es: ", suma)
#Enviamos el resultado de la suma por un nuevo topic 
respuesta = rospy.Publisher('respuesta_B', Int32, queue_size=10)
print("i es: ", i)


rospy.spin()

File: Boletines/C++/Boletin1/nodos (1)/src/nodoB.cpp
#include <ros/ros.h>
#include <std_msgs/Int32.h>
using namespace std;

//Definicion de 
int suma = 0;
int contador = 0;
ros::Publisher pub;

void callback(const std_msgs::Int32::ConstPtr& msg){
    /*Funcion que se ejecuta cuando se recibe un mensaje por el subcriptor nodoB*/
    ROS_INFO("EL numero recibido es %d", msg->data);

    //Defincion de variables globales
    int numero = msg->data; 

    //Creacion de la suma 
    suma += numero; ;
    contador++; 

    
    if (contador %10 == 0){
        //Creacion del mensaje 
        std_msgs::Int32 Mensaje;
        Mensaje.data = suma;
        pub.publish(Mensaje);
        //Reiniciamos el valor de suma 
        suma = 0 ;
        ROS_INFO("La suma es: %d", suma);
    }
    
}

int main(int argc, char **argv){
    //Creacion del nodo principal 
    ros::init(argc, argv, "NodoBPrincipal");
    ros::NodeHandle nh;

    //Creacion del subscriptor
    ros::Subscriber nodoB = nh.subscribe("nodoA", 10, callback);

    //Creaciond del publicador 
    pub = nh.advertise<std_msgs::Int32>("nodoB", 10);
    

    //Bucle 
    ros::spin();
    return 0;

}